File: Celebratorica/app/src/main/java/com/spiddekauga/celebratorica/item/CategoryEvent.java
package com.spiddekauga.celebratorica.item;

import com.spiddekauga.android.util.ObjectEvent;

import java.util.List;

/**
 * Category event
 */
class CategoryEvent extends ObjectEvent<Category> {

CategoryEvent(Category category, Actions action) {
	super(category, action);
}

CategoryEvent(List<Category> categories, Actions action) {
	super(categories, action);
}
}

File: remember-my-wins/app/src/main/java/io/blushine/rmw/item/CategoryEvent.kt
package io.blushine.rmw.item

import io.blushine.android.common.ObjectEvent

/**
 * Category event
 * @param movedFrom set if this is a move event
 * @param movedTo set if this is a move event
 */
internal class CategoryEvent @JvmOverloads constructor(action: Actions, items: List<Category> = listOf(), val movedFrom: Int = -1, val movedTo: Int = -1) : ObjectEvent<Category>(action, items) {
	constructor(action: ObjectEvent.Actions, category: Category) : this(action, listOf(category))

	fun isMoveEvent(): Boolean {
		return movedFrom != -1 && movedTo != -1
	}
}
